### ISSUE TITLE
Keeps freezing and unfreezing on Linux

### ISSUE THREAD

**Issue by gdelazzari** [vulkano-rs/vulkano#1136]
*   Version of vulkano: 0.11.1
*   OS: `Linux manjaro 4.19.6-1-MANJARO #1 SMP PREEMPT Sat Dec 1 12:21:26 UTC 2018 x86_64 GNU/Linux`
*   GPU (the selected PhysicalDevice): `GeForce GTX 660`
*   GPU Driver: `415.18`

# Issue
I'm sorry my explanation will be quite vague, but I'm learning Vulkan right now so I'm still not able to go in depth, but I can try my best if you guide me a bit.

I experience constant freezes and unfreezes with the triangle.rs example, i.e. 0.5/1 second after the window has opened my entire system freezes for 5/6 seconds (the mouse cursor is stuck), then unfreezes. If I click the titlebar of the window to move it (or try to resize it) a freeze occurs immediately. I *can* move and resize the window, it's just that it takes my PC almost a minute of freezing and unfreezing to accomplish the task. I do not notice any increase in RAM usage, my CPU usage however seems to go up quite a bit. Unfortunately I can't see how much and on how many cores because my PC is frozen while that happens, I just hear my fan spinning up and the CPU usage graph in the system monitor goes up a bit but I guess it misses a good chunk of measurements while the system is stuck, so I can't see what's really happening.

Building & running in release mode the triangle example doesn't change what happens.

Another thing I noticed while following the guide/tutorial on the website (which is great by the way, many thanks to whoever spent the time to write it) is that, as soon as I got the code working to draw the triangle in a framebuffer, transfer it to a CpuAccessibleBuffer, read the buffer on the CPU side and save the image as PNG with the image crate, the program was taking quite a lot of time to run, something like 2 seconds to execute the command buffer and 5 seconds to save the image, which I guess is weird. When compiled in release mode, however, it executed instantly as I expected.

As I said in the beginning, I'm sorry I can't better explain or investigate the issue since I know very little about Vulkan and Rust (I'm not new to graphics programming however, if that can help). If you need any specific test or want me to try any specific code I'll be happy to help. Thanks.

**Comment by glowing-chemist** [comment]
What desktop environment are you using? I experience small stalls on the desktop with gnome (3.30.2) . Your issue sounds like a driver/DE problem have you tried to reproduce on an older driver version?

**Comment by gdelazzari** [comment]
Unfortunately I had to replace my video card for various reasons (this happened like 3 days after I posted the issue) and I'm now on an RX580 with the mesa drivers, and now everything works fine... so it was indeed a driver issue (or a driver + DE one), not sure if I was the only one. I'm sorry I can't investigate further. Maybe this issue can be left open in case someone else sees the same behavior?
